moment_kinetics/src/gauss_legendre.jl

```diff
@@ -906,6 +913,26 @@ function setup_global_weak_form_matrix!(QQ_global::Array{mk_float,2},
             QQ_global[end,:] .= 0.0
             QQ_global[end,end] = 1.0
         end
+        # requires RHS vector b[1],b[end] = boundary values
+    end
+    if periodic_bc
+        # Make periodic boundary condition by modifying elements of matrix for duplicate point
+        # add assembly contribution to lower endpoint from upper endpoint
+        j = coord.nelement_local
+        get_QQ_local!(QQ_j,j,lobatto,radau,coord,option)
+        iminl = imin[j] - (coord.nelement_local > 1)
+        imaxl = imax[j]
+        QQ_global[1,iminl:imaxl] .+= QQ_j[end,:]

```

Comment:
This looks wrong to me, we should only add on to the points which overlap between elements, not the whole row. My previous line was correct.